### ISSUE TITLE
Evaluate the possibility to make PersistenceTimer running in multi threads

### ISSUE THREAD

**Issue by wu-sheng** [apache/skywalking#6472]
I am reviewing the core of OAP recently, one side to evaluate the requirements for the BanyanDB, and find the chance to improve the performance. #6411 is the first, this is the second one.

PersistenceTimer takes the core responsibility to assemble the requests and execute them to the storage. I noticed, the stage of assembly and execution is one-thread model. I remember these 2 stage causes many resources and latency. Considering the payload of OAP is usually not high, we could increase the number of thread to improve this.

@EvanLjp Could you share some stats data of self observability about these two?

FYI @apache/skywalking-committers

**Comment by EvanLjp** [comment]
![image](https://user-images.githubusercontent.com/31562192/109500783-a5012000-7ad1-11eb-8ae5-5aa3b5062cbc.png)
![image](https://user-images.githubusercontent.com/31562192/109500788-a7fc1080-7ad1-11eb-9e8f-3052aa10a523.png)

**Comment by wu-sheng** [comment]
@EvanLjp Back to you, have you tried to change `PersistenceTimer#extractDataAndSave`'s execution stage into the concurrency mode, such as using 2-4 threads to compare the performance? The 20s seem a little long.

**Comment by EvanLjp** [comment]
let me have a try.

**Comment by xuanyu66** [comment]
I want to konw, why metrics need to persist synchronously, can we just persist like record

**Comment by wu-sheng** [comment]
> @wu-sheng I want to konw, why metrics need to persist synchronously, can we just persist like record

Simple answer is, NO, they need to be in the sync way.. And we can't afford the load to explain how the source codes work.

**Comment by zifeihan** [comment]
@wu-sheng @xuanyu66 @EvanLjp I have also encountered the problem of es holding a lot of objects. in addition to multi-threaded writing, I also made an optimization here. I rewritten the buildBatchRequests method in MetricsPersistentWorker and added a parameter for batch writing. , Is to avoid frequent updates of hourly and day indicators. In addition, I made a time window to write one minute of data asynchronously (window period). If the current one minute has passed, then use synchronous writing. We have observed that the effect is good. hope some suggestions.
